android-sdk/src/androidTest/java/com/mobilecoin/lib/AccountTest.java

```diff
@@ -104,46 +142,100 @@ public void test_deserialize() throws SerializationException, InvalidUriExceptio
                 accountKey.getSubAddressSpendKey().getPublicKey()
         );
     }
-/* // TODO:
+
     @Test
-    public void test_account_integrity() throws SerializationException, BadEntropyException,
-            InvalidUriException, BadMnemonicException {
-        byte[] rootEntropy = new byte[32];
-        Arrays.fill(
-                rootEntropy,
-                (byte) 0
-        );
-        String mnemonic = Mnemonics.bip39EntropyToMnemonic(rootEntropy);
-        AccountKey accountKey = AccountKey.fromMnemonicPhrase(
-                mnemonic,
-                0,
-                fogConfig.getFogUri(),
-                fogConfig.getFogReportId(),
-                fogConfig.getFogAuthoritySpki()
-        );
+    public void fromMnemonicPhrase_createsCorrectPrivateKeys() throws Exception {
+        Assert.assertFalse(accountTestData.isEmpty());
+        for (AccountTestData accountTestData : accountTestData) {
+            String calculatedMnemonicPhrase =
+                    Mnemonics.bip39EntropyToMnemonic(accountTestData.bip39Entropy);
+            Assert.assertEquals(calculatedMnemonicPhrase, accountTestData.mnemonic);
```

Comment:
`assertEquals` treats the first argument as an "expected result", so if this test fails the tester will see an incorrect message, something like "expected value `XYZ`, but got `ABC`" while actually `ABC` is the correct value

Reply:
Done.